Makefile

```diff
@@ -1,6 +1,30 @@
 RIME_ROOT ?= $(CURDIR)
 
+ifeq ($(shell uname),Darwin) # for macOS
+prefix ?= $(RIME_ROOT)/dist
+
+ifdef BOOST_ROOT
+CMAKE_BOOST_OPTIONS = -DBoost_NO_BOOST_CMAKE=TRUE \
+	-DBOOST_ROOT="$(BOOST_ROOT)"
+endif
+
+# https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_SYSROOT.html
```

Comment:
@lotem I think the answer is yes. We can merge `Makefile` and `xcode.mk` since the latter just specify  `CMAKE_GENERATOR` as Xcode and nothing else is different with the former.